File: Linked LIst/Remove Duplicates from Linked List/RemoveDuplicate.cs
public class Program
{
    public class LinkedList
    {
        public int value { get; set; }
        public LinkedList next { get; set; }

        public LinkedList(int value)
        {
            this.value = value;
            this.next = null;
        }
    }

    // o(n) time complexity with o(1) space complexity
    public LinkedList RemoveDuplicateFromLinkedList(LinkedList list)
    {
        var currentNode = list;
        while (currentNode != null)
        {
            LinkedList nextDistinceNode = currentNode.next;
            while (nextDistinceNode != null && nextDistinceNode.value == currentNode.value)
            {
                nextDistinceNode = nextDistinceNode.next;
            }
            currentNode.next = nextDistinceNode;
            currentNode = nextDistinceNode;
        }
        return list;
    }
    public static void Main(string[] arg)
    {

    }
}

File: Linked LIst/Remove Duplicates from Linked List/RemoveDuplicate.js
class LinkedList {
    constructor(value){
        this.value = value;
        this.next = null;
    }
}

// o(n) time complexity with o(1) space complexity
function removeDuplicatesFromLinkedList(linkedList){
    let currentNode = linkedList;
    while(currentNode != null){
        let nextDistinctNOde = currentNode.next;
        while(nextDistinctNOde !== null && nextDistinctNOde.value === currentNode.value){
            nextDistinctNOde = nextDistinctNOde.next;
        }
        currentNode.next = nextDistinctNOde;
        currentNode = nextDistinctNOde;
    }
    return linkedList;
}


